app/models/legal_aid_application.rb

```diff
@@ -446,6 +446,62 @@ def family_linked_associated_application?
     lead_linked_application&.link_type_code == "FC_LEAD"
   end
 
+  def legal_linked_associated_application?
+    lead_linked_application&.link_type_code == "LEGAL"
+  end
+
+  def family_linked?
+    @family_linked ||= family_linked_applications_count.positive?
+  end
+
+  def legal_linked?
+    @legal_linked ||= legal_linked_applications_count.positive?
+  end
+
+  def family_linked_lead_or_associated
+    return "" unless family_linked?
+
+    @family_linked_lead_or_associated ||= family_linked_associated_application? ? "Associated" : "Lead"
+  end
+
+  def legal_linked_lead_or_associated
+    return "" unless legal_linked?
+
+    @legal_linked_lead_or_associated ||= legal_linked_associated_application? ? "Associated" : "Lead"
+  end
+
+  def family_linked_applications_count
+    @family_linked_applications_count ||= family_linked_associated_application? ? associated_family_linked_count : lead_family_linked_count
+  end
+
+  def legal_linked_applications_count
+    @legal_linked_applications_count ||= legal_linked_associated_application? ? associated_legal_linked_count : lead_legal_linked_count
+  end
+
+  def associated_family_linked_count
+    return 0 unless lead_application
```

Comment:
I'm assuming these `return` statements could be removed if a safe navigation operator was added below: 

```
lead_application.associated_linked_applications&.where(link_type_code: "FC_LEAD", confirm_link: true)&.count
```

A personal preference thing I guess and I'm happy with either tbh

Reply:
Removed as part of refactor (although I believe safe navigation would have returned nil instead of 0)